Python/modsupport.c

```diff
@@ -678,3 +678,20 @@ PyModule_AddStringConstant(PyObject *m, const char *name, const char *value)
     Py_DECREF(o);
     return -1;
 }
+
+int
+PyModule_AddType(PyObject *module, PyTypeObject *type)
+{
+    if (PyType_Ready(type) < 0) {
+        return -1;
+    }
+
+    const char *name = _PyType_Name(type);
+    Py_INCREF(type);
```

Comment:
```suggestion
    const char *name = _PyType_Name(type);
    assert(name != NULL);
    
    Py_INCREF(type);
```

Please add also `assert(type->tp_name != NULL);` at the beginning of `_PyType_Name()`.